lib/active_merchant/billing/gateways/cecabank/cecabank_json.rb

```diff
@@ -0,0 +1,217 @@
+        build_cancellation(:refund, money, authorization, options)
+      end
+
+      private
+
+      def build_purchase(action, money, creditcard, options)
+        requires!(options, :order_id)
+
+        post = { parametros: { accion: CECA_ACTIONS_DICTIONARY[action] } }
+
+        add_invoice(post, money, options)
+        add_creditcard(post, creditcard)
+        add_stored_credentials(post, creditcard, options)
+        add_three_d_secure(post, options)
+
+        commit('compra', post)
+      end
+
+      def build_cancellation(action, money, authorization, options = {})
+        post = {}
+        add_auth_invoice_data(action, post, money, authorization, options)
+
+        commit('anulacion', post)
+      end
+
+      def add_auth_invoice_data(action, post, money, authorization, options)
+        requires!(options, :order_id)
+
+        params = post[:parametros] ||= {}
+        params[:accion] = CECA_ACTIONS_DICTIONARY[action]
+        params[:referencia] = authorization
+
+        add_invoice(post, money, options)
+      end
+
+      def add_encryption(post)
+        post[:cifrado] = CECA_ENCRIPTION
+      end
+
+      def add_signature(post, params_encoded, options)
+        post[:firma] = Digest::SHA2.hexdigest(@options[:signature_key].to_s + params_encoded)
+      end
+
+      def add_merchant_data(post)
+        params = post[:parametros] ||= {}
+
+        params[:merchantID] = @options[:merchant_id]
+        params[:acquirerBIN] = @options[:acquirer_bin]
+        params[:terminalID] = @options[:terminal_id]
+      end
+
+      def add_invoice(post, money, options)
+        params = post[:parametros] ||= {}
+
+        params[:numOperacion] = options[:order_id]
+        params[:importe] = amount(money)
+        params[:tipoMoneda] = CECA_CURRENCIES_DICTIONARY[options[:currency] || currency(money)].to_s
+        params[:exponente] = 2.to_s
+      end
+
+      def add_creditcard(post, creditcard)
+        params = post[:parametros] ||= {}
+
+        params[:pan] = creditcard.number
+        params[:caducidad] = year_format(creditcard)
+        params[:cvv2] = creditcard.verification_value
+        params[:csc] = creditcard.verification_value if CreditCard.brand?(creditcard.number) == 'american_express'
+      end
+
+      def add_stored_credentials(post, creditcard, options)
+        params = post[:parametros] ||= {}
+        return if options[:exemption_sca] == 'NONE'
+
+        params[:exencionSCA] = options[:exemption_sca]
+        requires!(options, :reason_type, :initiator, :days_frequency) if options[:exemption_sca] == 'MIT'
+
+        params[:tipoCOF] = options[:reason_type]
+        params[:inicioRec] = options[:initiator]
+        params[:finRec] = options[:expiration_date] || year_format(creditcard)
+        params[:frecRec] = options[:days_frequency]
+
+        params[:mmppTxId] = options[:network_transaction_id] if options[:network_transaction_id]
+      end
+
+      def add_three_d_secure(post, options)
```

Comment:
This change is temporary, it is necessary to test authorize and purchase, we have some tickets to implement 3DS, for now I am going to leave it but it will be modified in the global 3DS implementation